pkg/dashboard/resource/apigateway.go

```diff
@@ -174,13 +176,47 @@ func (agr *apiGatewayResource) updateAPIGateway(request *http.Request) (*restful
 		ValidateFunctionsExistence: agr.headerValueIsTrue(request, "x-nuclio-agw-validate-functions-existence"),
 	}); err != nil {
 		agr.Logger.WarnWithCtx(ctx, "Failed to update api gateway", "err", err)
+		return nil, nuclio.WrapErrInternalServerError(errors.Wrap(err, "Failed to update api gateway"))
```

Comment:
why do you wrap those errors with error with http? 
note that on nuclio, we solve the _top_ error with http status as the http status that returned to the user. so basically, you will mask any error returned by `UpdateAPIGateway` (such as forbidder / invalid body / etc).